source/base/data_out_base.cc

```diff
@@ -7741,19 +7748,49 @@ DataOutInterface<dim, spacedim>::write_vtu_in_parallel(
                                   vtk_flags,
                                   ss);
 
-    ierr = MPI_File_write_ordered(
-      fh, ss.str().c_str(), ss.str().size(), MPI_CHAR, MPI_STATUS_IGNORE);
+    // using prefix sum to find specific offset to write at.
+
+    const std::uint64_t size_on_proc = ss.str().size();
+    size_placeholder = size_on_proc;
+    std::uint64_t prefix_sum = 0;
+    ierr = MPI_Exscan(&size_on_proc, &prefix_sum, 1, MPI_CHAR, MPI_SUM, comm);
```

Comment:
The MPI type is incorrect. What type is the argument with the data to this MPI call?

Reply:
Ohh I see, it should be MPI_UINT64_T right?

Reply:
correct.